Remove an iterator to avoid serialziation errors

wadells opened the pull request:
## Description
The master Gravity CI builds on Jenkins are intermittently hitting errors like the following:

```
Caused: java.io.NotSerializableException: java.util.Collections$UnmodifiableMap$UnmodifiableEntrySet$UnmodifiableEntry
```

These manifiest as the  'unit test gravity' stage failing in <1s.

This is an attempt to avoid them by removing an iterator (which is non-serializable)

See: https://www.jenkins.io/blog/2017/02/01/p.ipeline-scalability-best-practice/#appendix-serializable-vs.-non-serializable-types and https://stackoverflow.com/a/40456164

Major kudos to @a-palchikov for several key hints on the way to debugging this.


## Type of change
* Bug fix (non-breaking change which fixes an issue)
* Internal change (not necessarily a bug fix or a new feature)

## Linked tickets and other PRs
* Fixes https://github.com/gravitational/gravity/issues/2227

## TODOs
- [x] Self-review the change
- [x] Perform manual testing
- [x] Address review feedback

## Testing done
I am re-runing this PR build several times to see if it hits the failure which would intermittently occur on other PRs.  Compare:
* This PR's history: https://jenkins.gravitational.io/job/Gravity-PR-5.2/job/PR-2223/  After the iterator fix (build 8+) we have a 100% success rate.  Given a ~50% probability of failure, there is <1% chance for 7 consecutive successes.
* A parallel PR without the change: https://github.com/gravitational/gravity/pull/2212 (~50% success rate, with unrelated unittest truncation)

a-palchikov commented on Jenkinsfile:
```diff
@@ -54,17 +54,16 @@ def robotest() {
       propagateParamsToEnv()
     }
   }
-  runRobotest = (env.RUN_ROBOTEST == 'run')
```
Is this just a guess since I thought booleans were [serializable](https://www.jenkins.io/blog/2017/02/01/pipeline-scalability-best-practice/#appendix-serializable-vs.-non-serializable-types) in groovy?

wadells replied:
Yeah, this was a hunch.  Looks like it didn't pan out:

https://jenkins.gravitational.io/job/Gravity-PR-5.2/job/PR-2223/5/console

wadells committed:
Switch from iterator to for loop.

Iterators are not serializable, as discussed at:

  https://www.jenkins.io/blog/2017/02/01/pipeline-scalability-best-practice/#appendix-serializable-vs.-non-serializable-types

wadells commented:
> this could be a good candidate to move to an external function which could be marked @noncps

I agree.  I may eventually do this.  For now, I'm still iterating through hypotheses as to what the root cause is.  Once I've identified that, @noncps is a good way to quarantine the offending logic (and also leave a hint for future developers about this risk).

wadells committed:
Merge branch 'master' into walt/master-jenkinsfile-nonserializable
